source/simulator/core.cc

```diff
@@ -1250,57 +1230,11 @@ namespace aspect
                mpi_communicator);
 #endif
 
-    if ((parameters.use_discontinuous_temperature_discretization) || (parameters.use_discontinuous_composition_discretization))
-      {
-        Table<2,DoFTools::Coupling> face_coupling (introspection.n_components,
-                                                   introspection.n_components);
-        face_coupling.fill (DoFTools::none);
-
-        const typename Introspection<dim>::ComponentIndices &x
-          = introspection.component_indices;
-        if (parameters.use_discontinuous_temperature_discretization)
-          face_coupling[x.temperature][x.temperature] = DoFTools::always;
-
-        if (parameters.use_discontinuous_composition_discretization)
-          {
-            for (unsigned int c=0; c<introspection.n_compositional_fields; ++c)
-              {
-                const AdvectionField adv_field (AdvectionField::composition(c));
-                const typename Parameters<dim>::AdvectionFieldMethod::Kind method = adv_field.advection_method(introspection);
-                switch (method)
-                  {
-                    case Parameters<dim>::AdvectionFieldMethod::fem_field:
-                      face_coupling[x.compositional_fields[c]][x.compositional_fields[c]] = DoFTools::always;
-                      break;
-
-                    case Parameters<dim>::AdvectionFieldMethod::particles:
-                      face_coupling[x.compositional_fields[c]][x.compositional_fields[c]] = DoFTools::none;
-                      break;
-
-                    case Parameters<dim>::AdvectionFieldMethod::static_field:
-                      face_coupling[x.compositional_fields[c]][x.compositional_fields[c]] = DoFTools::none;
-                      break;
-
-                    default:
-                      Assert(false,ExcNotImplemented());
-                  }
-              }
-          }
-
-        DoFTools::make_flux_sparsity_pattern (dof_handler,
```

Comment:
How come we don't need to call this function any more? Aren't there cases where we need these entries? Or is this just in the system matrix, not the preconditioner? And what do we use for a preconditioner when we solve temperature/composition?